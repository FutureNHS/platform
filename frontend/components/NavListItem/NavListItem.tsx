import React from "react";

import Link from "next/link";
import styled from "styled-components";

import { Folder } from "../../lib/generated/graphql";

type ListItem = Pick<Folder, "id" | "title">;

interface Props {
  active: boolean;
  item: ListItem;
  workspaceId: string;
  imgSrc: string;
}

const ListItem = styled.li<{ active: boolean }>`
  background: ${({ active, theme }) =>
    active ? theme.colorNhsukYellow : "inherit"};
  border-radius: 4px;
  font-weight: ${({ active }) => (active ? 700 : "inherit")};
  list-style-type: none;
  margin-bottom: 12px;
  &:hover {
    ${({ theme }) => `
    background-color: ${theme.colorNhsukGrey4}
  `}
  }
  a {
    display: flex;
    padding-left: 8px;
    text-decoration: none;
    &:focus {
      box-shadow: none;
    }
  }
  img {
    padding-top: 2px;
    align-self: baseline;
    /* stop things jumping about while svgs load */
    width: 24px;
    flex-shrink: 0;
  }
  div {
    padding-left: 4px;
    padding-right: 4px;
    ${({ theme }) => `
    color: ${theme.colorNhsukBlack};
  `}
  }
`;

const NavListItem = ({ active, item, workspaceId, imgSrc }: Props) => (
  <ListItem active={active}>
    <Link href={`/workspaces/${workspaceId}/folders/${item.id}`}>
      <a>
        <img src={imgSrc} />
        <div>{item.title}</div>
      </a>
    </Link>
  </ListItem>
);

export default NavListItem;
