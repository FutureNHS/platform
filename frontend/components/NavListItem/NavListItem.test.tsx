import React from "react";

import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import theme from "../../lib/fixtures/theme.json";
import NavListItem from "./NavListItem";

describe("<NavListItem/>", () => {
  const folder = { id: "folder-id", title: "folder title" };

  it("renders matching snapshot when not active", () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <NavListItem
          active={false}
          item={folder}
          href="test-link"
          imgSrc="test-svg"
        />
      </ThemeProvider>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });

  it("renders matching snapshot when active", () => {
    const container = render(
      <ThemeProvider theme={theme}>
        <NavListItem
          active={true}
          item={folder}
          href="test-link"
          imgSrc="test-svg"
        />
      </ThemeProvider>
    );

    expect(container.asFragment()).toMatchSnapshot();
  });
});
