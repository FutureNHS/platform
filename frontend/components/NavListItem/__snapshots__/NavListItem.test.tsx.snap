// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavListItem/> renders matching snapshot when active 1`] = `
<DocumentFragment>
  .c0 {
  background: rgb(255,235,59);
  border-radius: 4px;
  font-weight: 700;
  list-style-type: none;
  margin-bottom: 12px;
}

.c0:hover {
  background-color: rgb(216,221,224);
}

.c0 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:focus {
  box-shadow: none;
}

.c0 img {
  padding-top: 2px;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  width: 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 div {
  padding-left: 4px;
  padding-right: 4px;
  color: rgb(33,43,50);
}

<li
    class="c0"
  >
    <a
      href="/test-link"
    >
      <img
        src="test-svg"
      />
      <div>
        folder title
      </div>
    </a>
  </li>
</DocumentFragment>
`;

exports[`<NavListItem/> renders matching snapshot when not active 1`] = `
<DocumentFragment>
  .c0 {
  background: inherit;
  border-radius: 4px;
  font-weight: inherit;
  list-style-type: none;
  margin-bottom: 12px;
}

.c0:hover {
  background-color: rgb(216,221,224);
}

.c0 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 8px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 a:focus {
  box-shadow: none;
}

.c0 img {
  padding-top: 2px;
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  width: 24px;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c0 div {
  padding-left: 4px;
  padding-right: 4px;
  color: rgb(33,43,50);
}

<li
    class="c0"
  >
    <a
      href="/test-link"
    >
      <img
        src="test-svg"
      />
      <div>
        folder title
      </div>
    </a>
  </li>
</DocumentFragment>
`;
